shell/server/proxy.js

```diff
@@ -36,6 +36,85 @@ SANDSTORM_ALTHOME = Meteor.settings && Meteor.settings.home;
 SANDSTORM_LOGDIR = (SANDSTORM_ALTHOME || "") + "/var/log";
 SANDSTORM_VARDIR = (SANDSTORM_ALTHOME || "") + "/var/sandstorm";
 
+var storeReferralProgramInfoApiTokenCreated = function(db, accountId, identityId, apiTokenAccountId) {
+  // From the Referral program's perspective, if Bob's Account has no referredByComplete, then we
+  // update Bob's Identity to say it's referredBy Alice's Account (which is apiTokenAccountId).
+  check(accountId, String);
+  check(identityId, String);
+  check(apiTokenAccountId, String);
+
+  // Bail out early if quota enforcement is disabled.
+  if (! Meteor.settings.public.quotaEnabled) {
+    return;
+  }
+
+  var aliceAccountId = apiTokenAccountId;
+  var bobAccountId = accountId;
+  var bobIdentityId = identityId;
+
+  if (Meteor.users.find({_id: bobAccountId,
+                         referredByComplete: {$exists: true}}).count() > 0) {
+    return;
+  }
+
+  // Only actually update Bob's Identity ID if there is no referredBy.
+  var updatedCount = Meteor.users.update(
```

Comment:
FWIW in a more heavily `console.log`-y version of this code, I used to log `updatedCount`. Agreed that the variable is superflous; fixed in a follow-up commit.